libse/Dictionaries/NameList.cs

```diff
@@ -282,12 +282,24 @@ public bool IsInNamesMultiWordList(string input, string word)
 
         public bool ContainsCaseInsensitive(string name)
         {
-            if (string.IsNullOrEmpty(name))
+            if (name == null)
             {
                 return false;
             }
 
-            foreach (var n in name.Contains(' ') ? _namesMultiList : _namesList)
+            // name must contain atleast two chars
+            if (name.Length < 2)
+            {
+                return false;
+            }
+
+            var lookupCollection = name.Contains(' ') ? _namesMultiList : _namesList;
+            if (lookupCollection.Contains(name)) // O(1)
```

Comment:
will save of from running O(n) when `name` exists in `name-list`  w/ right casing